src/i2c/README.md

```diff
@@ -1,41 +1,64 @@
-# Crate Name
+# vhost-device-i2c - I2C emulation backend daemon
 
-## Design
+## Description
+This program is a vhost-user backend that emulates a VirtIO I2C bus.
+This program takes the layout of the i2c bus and its devices on the host
+OS and then talks to them via the /dev/i2c-X interface when a request
+comes from the guest OS for an I2C or SMBUS device.
 
-TODO: This section should have a high-level design of the crate.
+This program is tested with QEMU's `-device vhost-user-i2c-pci` but should
+work with any virtual machine monitor (VMM) that supports vhost-user. See the
+Examples section below.
 
-Some questions that might help in writing this section:
-- What is the purpose of this crate?
-- What are the main components of the crate? How do they interact which each
-  other?
+## Synopsis
 
-## Usage
+**vhost-device-i2c** [*OPTIONS*]
 
-TODO: This section describes how the crate is used.
+## Options
 
-Some questions that might help in writing this section:
-- What traits do users need to implement?
-- Does the crate have any default/optional features? What is each feature
-  doing?
-- Is this crate used by other rust-vmm components? If yes, how?
+.. program:: vhost-device-i2c
+
+.. option:: -h, --help
+
+  Print help.
+
+.. option:: -s, --socket-path=PATH
+
+  Location of vhost-user Unix domain sockets, this path will be suffixed with
+  0,1,2..socket_count-1.
+
+.. option:: -c, --socket-count=INT
+
+  Number of guests (sockets) to attach to, default set to 1.
+
+.. option:: -l, --device-list=I2C-DEVICES
+
+  I2c device list at the host OS in the format:
+      <bus>:<client_addr>[:<client_addr>],[<bus>:<client_addr>[:<client_addr>]]
+
+      Example: --device-list "2:32:21,3:10:23"
+
+  Here,
+      bus (decimal): adatper bus number. e.g. 2 for /dev/i2c-2, 3 for /dev/i2c-3.
+      client_addr (decimal): address for client device, 32 == 0x20.
 
 ## Examples
 
-TODO: Usage examples.
+The daemon should be started first:
+
+::
 
-```rust
-use my_crate;
+  host# vhost-device-i2c --socket-path=vi2c.sock --socket-count=1 --device-list 0:32
```

Comment:
Can this be an integration test?

Reply:
> Can this be an integration test?

The kind of sample tests that I was able to find tested the routines directly. What you are suggesting here is a bit different, test the generated binary. Can you point me to some sample code for this kind of testing ?

Reply:
We are doing tests on the binary in vmm-reference. Here is an example of a test: https://github.com/rust-vmm/vmm-reference/blob/1226002a93a4dd857864ec4bc2eed3688e9870e3/tests/test_run_reference_vmm.py#L296.  We're currently testing the vmm-reference binary using Python integration tests, but in this case we should see if writing Rust integration tests can be implemented. We can open a separate issue for this one.

Reply:
> We are doing tests on the binary in vmm-reference. Here is an example of a test: https://github.com/rust-vmm/vmm-reference/blob/1226002a93a4dd857864ec4bc2eed3688e9870e3/tests/test_run_reference_vmm.py#L296. We're currently testing the vmm-reference binary using Python integration tests, but in this case we should see if writing Rust integration tests can be implemented. We can open a separate issue for this one.

I fixed it differently, broke most of main.rs into another routine which I am able to test now. Cover is 44% now.